crma_api_client/resources/dataset.py

```diff
@@ -16,6 +17,114 @@ class Dataset(BaseModel):
     url: str
 
 
+class DateType(Enum):
+    """The type of date field"""
+
+    date = "Date"
+    date_only = "DateOnly"
+    date_time = "DateTime"
+
+
+class XmdDateFields(BaseModel):
+    """Date fields in the extended metadata format
+
+    See https://developer.salesforce.com/docs/atlas.en-us.230.0.bi_dev_guide_rest.meta/bi_dev_guide_rest/bi_resources_xmd_main.htm#XmdDateFieldRepresentation
+    """
+
+    day: str
+    epoch_day: str
+    epoch_second: str
+    fiscal_month: Optional[str]
+    fiscal_quarter: Optional[str]
+    fiscal_week: Optional[str]
+    fiscal_year: Optional[str]
+    full_field: str
+    hour: str
+    minute: str
+    month: str
+    quarter: str
+    second: str
+    week: str
+    year: str
+
+    @validator("fiscal_month", "fiscal_quarter", "fiscal_week", "fiscal_year")
+    def not_none(cls, v):
+        """The value should not be None"""
+        if v is None:
+            raise ValueError("Fiscal fields may not be None")
+        return v
+
+    class Config:
+        """Model configuration"""
+
+        alias_generator = to_camel
```

Comment:
is this `Config` class a standard pydantic convention? or is this just something we do?

feel free to ignore this, but would it be helpful to have a mixin or something that adds this `Config` class to each of these pydantic models? it looks like the `Config` class is the same on each one

Reply:
It's a pydantic thing. I tried using a mixin but pydantic wasn't picking it up. Let me try one more thing